src/main/scala/carldata/series/Analysis.scala

```diff
@@ -0,0 +1,38 @@
+package carldata.series
+
+import scala.annotation.tailrec
+
+object Analysis {
+  implicit class SeriesAnalysis[V](val ts: TimeSeries[V])(implicit num: Numeric[V]) {
+
+    def findSessions: Seq[Session] = {
+      @tailrec def traverseTimeSeries[V](ts: TimeSeries[V],
```

Comment:
It would be much cleaner (and simpler) to use zip with foldL instead of this recursive function